Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -68,34 +81,234 @@ sendPaymentReminderToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId
+    _ -> case transporterConfig.driverFeeCalculatorBatchGap of
+      Nothing -> throwError $ InternalError "No batch gap defined for driver fee calculator job"
+      Just gap -> ReSchedule <$> getRescheduledTime gap
+  where
+    mkApplyOfferRequest offerTxnUUID appliedOfferIds due driverPlan' dutyDate registrationDate =
+      PaymentInterface.OfferApplyReq
+        { txnId = offerTxnUUID,
+          offers = appliedOfferIds,
+          customerId = driverPlan'.driverId.getId,
+          amount = due,
+          currency = PaymentInterface.INR,
+          planId = driverPlan'.planId.getId,
+          registrationDate,
+          dutyDate = dutyDate,
+          paymentMode = show $ driverPlan'.planType
+        }
+
+buildRestFees :: MonadGuid m => DriverFeeStatus -> FeeType -> DriverFee -> m DriverFee
+buildRestFees status_ feeType_ DriverFee {..} = do
+  id_ <- generateGUID
+  return
+    DriverFee
+      { id = id_,
+        status = status_,
+        feeType = feeType_,
+        ..
+      }
+
+calculatePlatformFeeAttr :: HighPrecMoney -> Plan -> (HighPrecMoney, HighPrecMoney, HighPrecMoney)
+calculatePlatformFeeAttr totalFee plan = do
+  let platformFee = totalFee / HighPrecMoney (toRational $ 1 + plan.cgstPercentage + plan.sgstPercentage) -- this should be changed to HighPrecMoney
+      cgst = HighPrecMoney (toRational plan.cgstPercentage) * platformFee
+      sgst = HighPrecMoney (toRational plan.sgstPercentage) * platformFee
+  (platformFee, cgst, sgst)
+
+makeOfferReq :: HighPrecMoney -> Person -> Plan -> UTCTime -> UTCTime -> Payment.OfferListReq
+makeOfferReq totalFee driver plan dutyDate registrationDate = do
+  let offerOrder = Payment.OfferOrder {orderId = Nothing, amount = totalFee, currency = Payment.INR} -- add UDFs
+      customerReq = Payment.OfferCustomer {customerId = driver.id.getId, email = driver.email, mobile = Nothing}
+  Payment.OfferListReq
+    { order = offerOrder,
+      customer = Just customerReq,
+      planId = plan.id.getId,
+      registrationDate,
+      paymentMode = show plan.paymentMode,
+      dutyDate
+    }
+
+getFinalOrderAmount :: (EncFlow m r, CacheFlow m r, EsqDBFlow m r) => HighPrecMoney -> Id Merchant -> TransporterConfig -> Person -> Plan -> HighPrecMoney -> UTCTime -> DriverFee -> m (HighPrecMoney, HighPrecMoney, Maybe Text, Maybe Text)
+getFinalOrderAmount feeWithoutDiscount merchantId transporterConfig driver plan baseAmount registrationDate driverFee = do
+  now <- getCurrentTime
+  let dutyDate = addUTCTime (secondsToNominalDiffTime transporterConfig.timeDiffFromUtc) driverFee.createdAt
+      registrationDateLocal = addUTCTime (secondsToNominalDiffTime transporterConfig.timeDiffFromUtc) registrationDate
+  if feeWithoutDiscount == 0
+    then do
+      updateStatus CLEARED now driverFee.id
+      return (0, 0, Nothing, Nothing)
+    else do
+      offers <- Payment.offerList merchantId (makeOfferReq feeWithoutDiscount driver plan dutyDate registrationDateLocal) -- handle UDFs
```

Comment:
Lets add try, catch here. In case of error, lets not process that driverFee here.